### PR TITLE
Fix conflation of empty and nil slice for routes, in app update handling

### LINKED ISSUE #1799: Remove all the routes from an application
A user can remove all the routes from an application.

**Is your feature request related to a problem? Please describe.**
Removing the routes from the UI will not work, because is not handled on the BE side.

**Describe the solution you'd like**
I would like to delete the routes from a deployed application.

**Describe alternatives you've considered**

**Additional context**

- https://github.com/epinio/epinio/issues/1775

### PR DESCRIPTION
**Pull request by andreas-kupries**
Fix #1799 

nil   => no change
empty => clear routes

Feat: extend cli with ability send signal for clearing routes (new update-only option --clear-routes, -z)

### CODE REVIEW

**Comment by andreas-kupries** [comment]
The single failure is a plain push with custom secret - Different from first run, test for overcoming a bad staging.
I suspect that these are transient network and cluster issues. Local execution of these tests is fine.

File: acceptance/apps_test.go [hunk 1]
```diff
@@ -571,6 +631,8 @@ configuration:
 					return err
 				}
 
+				By("ZZZ JSON: ((" + statusJSON + "))")
+
```
Comment (enrichman) [inline acceptance/apps_test.go:635]: I guess this is a leftover.

Just FYI I've found the `GinkgoWriter` pretty handy to log things while testing. It has some convenience methods such as `GinkgoWriter.Println` and so on.

File: internal/cli/usercmd/push.go [hunk 2]
```diff
@@ -261,6 +257,24 @@ func (c *EpinioClient) stageLogs(appRef models.AppRef, stageID string) {
 	}()
 }
 
+func routeMessage(msg *termui.Message, routes []string) *termui.Message {
+	if routes == nil {
+		if len(routes) > 0 {
```
Comment (enrichman) [inline internal/cli/usercmd/push.go:262]: Hm, I'm not sure. How can routes be nil and `>0`?

**Review by enrichman — COMMENTED** [review]
Just a couple of comments.

File: acceptance/apps_test.go (hunk 1 above)
Comment (andreas-kupries) [reply acceptance/apps_test.go:635]: Yes. :(

File: internal/cli/usercmd/push.go (hunk 2 above)
Comment (andreas-kupries) [reply internal/cli/usercmd/push.go:262]: Will recheck the logic there.

Comment (andreas-kupries) [reply internal/cli/usercmd/push.go]: Fixed. Commit dd20c7680623195b1487aa257ebed8cb0a8824a2

The logic was inverted. Reworked to reduce indentation and make logic clearer.
This should have a full run of CI to see if there are tests affected by this change to the output.

Comment (enrichman) [reply internal/cli/usercmd/push.go]: Thank you, much better! :+1:

**Comment by enrichman** [comment]
LGTM, let's wait for CI :+1:

**Review by enrichman — APPROVED** [review]
@andreas-kupries LGTM, we need to remember to update the CLI docs :+1:

**Pull request merged**